swarm/storage/mru/resource_test.go

```diff
@@ -694,12 +700,14 @@ func TestValidator(t *testing.T) {
 	defer cancel()
 	startTime := rh.getCurrentTime(ctx)
 
-	chunk, _ = rh.newMetaChunk(&resourceMetadata{
+	metadata := &resourceMetadata{
 		name:      safeName,
 		startTime: startTime,
 		frequency: resourceFrequency,
 		ownerAddr: signer.Address(),
-	})
+	}
+	chunk, _ = metadata.newChunk()
```

Comment:
Same here. Generally the regime used in our codebase go more towards creating param structs passed to methods. Here it's the other way around. What is the advantage of breaking with our convention?

Reply:
This is not what I have seen in Marshallers and Unmarshallers, whereupon you call `.Marshall()` of a struct and get a byte array, or instantiate a struct and call `.Unmarshall(byte array)` to populate it. This pattern is for example in `resource.go` for the `resource` struct (`MarshallBinary`, `UnmarshallBinary`). Also in general go libraries, for example `big.Int`.

`newChunk` is a serializer that takes the metadata and gives you a serialized chunk, so my reasoning is it is a serializer, so it would make sense to construct it this way.

Another reasoning is that it allows me to group logically the `newChunk()` method as part of the `resourceMetadata` struct. Otherwise it would be kind of orphan.

In any case, please note that in the tests, we always hack around the internals of a package to test certain things, so I understand the construct looks verbose or out of place. But look at how clean it looks `metadata.newChunk()` where it is used in real code. :-)

Reply:
You're right in that it's only the more complex objects that we use the params setup.

If these are indeed (binary) Marshal and Unmarshal operations, they should implement those interfaces. In any case I think we should avoid arbitrary naming of serializers; it easily gets confusing. I've had this discussion with @janos before, and I think even we had a roundtable partially dedicated to the topic.

I'm sure there are examples in the code (maybe even _especially_ in mine) that suggest otherwise,

Reply:
In the last review, you will see that all serializers have a homogenized names. It will be easy to change them to whatever naming convention.

In the case of `newChunk` the exception would be that it does not return a byte array, but a ready to use `storage.Chunk`